Sources/Rendering/Core/ColorTransferFunction/index.js

```diff
@@ -214,22 +214,29 @@ function vtkColorTransferFunction(publicAPI, model) {
   // Set nodes directly
   publicAPI.setNodes = (nodes) => {
     if (model.nodes !== nodes) {
+      const before = JSON.stringify(model.nodes);
       model.nodes = nodes;
-      publicAPI.sortAndUpdateRange();
+      const after = JSON.stringify(model.nodes);
+      return publicAPI.sortAndUpdateRange() || before !== after;
```

Comment:
shouldn't `after` be set AFTER sortAndUpdateRange() is called ?

what if calling `setNodes([0, 2, 1])` with `model.nodes = [0, 1, 2]` ? `modified()` should NOT be called and false be returned. This is not the case here.

Reply:
I see the idea, but the result would be the same as the `sortAndUpdateRange()` would have then triggered a modified(). 

This mostly fixed the intended use case when nodes provided have the same content/layout as the one stored in the instance. We could push to support unsorted nodes and skip the modified, but so far I don't see the use case.

Reply:
I see your point. It will be a hard to find bug in case such unlikely scenario happens.
But I do not have a quick and elegant solution (I'm actually concerned that `stringify` gets costly in case of large CTF)  to suggest so I guess it's fine to leave it as is.

Reply:
modified() is not that evil. But once you have large meshes, you see the performance hit when GL structures get rebuilt. After it is just a matter of figuring which object mtime trigger the rebuild. Once you have a use case highlighting the issue, it is easy to optimize it or suggest changes on the app side to properly get the intended behavior. That issue never arose before, until I was dealing with large meshes with the SyncView.